cohttp-eio/src/cohttp_eio.mli

```diff
@@ -90,3 +92,75 @@ module Server : sig
 
   val not_found_handler : handler
 end
+
+module Client : sig
+  type response = Http.Response.t * Eio.Buf_read.t
+  type env = < net : Eio.Net.t >
+
+  type 'a body_disallowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    (< env ; .. > as 'a) ->
+    Eio.Switch.t ->
+    Eio.Net.Sockaddr.stream ->
+    Uri.t ->
+    response
+  (** [body_disallowed_call] denotes HTTP client calls where a request is not
+      allowed to have a request body. *)
+
+  type 'a body_allowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    ?body:Body.t ->
+    (< env ; .. > as 'a) ->
```

Comment:
Does it need to be passed as an object? I meant that the argument would be of type `Eio.Net.t`

Reply:
Yes, at the moment, the object just needs `net`. However, this object will likely have `sys_clock`, `mono_clock` and `secure_random` to accomodate ocaml-dns, ocaml-tls dependency requirements.